### PR TITLE
Add carousel for landing chapters section

### PR DESCRIPTION
**Pull request by grzegorz-s**
With new requirements, there is a need to create an option to display all chapters on the landing page. 

It's optional so you need pass specific flag when registering component in your Vuepress project and it has backward compatibility

### CODE REVIEW

File: components/ChapterBlock.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
+<template>
+  <div
+    data-aos="fade-up"
+    data-aos-duration="1000"
+    data-aos-offset="-200"
+    :data-aos-delay="200 * (order + 1)"
```
Comment (wooojek) [inline components/ChapterBlock.vue:6]: how about moving those numbers to some constants/variables? naming would provide at least some description for those :)

File: components/ChapterBlock.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      type: Number,
+      required: true,
+    },
+    hasCarousel: {
+      type: Boolean,
+      default: false,
+    },
+    limitSectionInChapter: {
+      type: Number,
+      default: undefined,
```
Comment (wooojek) [inline components/ChapterBlock.vue:58]: maybe align default value with type?

File: components/ChapterBlock.vue [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+export default {
+  components: {
+    ArrowRight,
+  },
+
+  props: {
+    order: {
+      type: Number,
+      required: true,
```
Comment (wooojek) [inline components/ChapterBlock.vue:50]: type mismatch

File: components/ChapterBlocks.vue [hunk 4]
```diff
@@ -1,48 +1,63 @@
...
+      >
+        <ArrowRight />
+      </button>
+    </div>
+  </div>
 </template>
 
 <script>
-import ArrowRight from '@theme/assets/svg/landing-arrow-right.svg'
+import Siema from 'siema'
```
Comment (wooojek) [inline components/ChapterBlocks.vue:51]: nice! was always wondering if I see it in some project 😄

File: components/ChapterBlocks.vue [hunk 5]
```diff
@@ -54,13 +69,54 @@ export default {
       type: Number,
       default: undefined,
```
Comment (wooojek) [inline components/ChapterBlocks.vue:70]: type mismatch

File: components/ChapterBlocks.vue [hunk 6]
```diff
@@ -54,13 +69,54 @@ export default {
...
+
+  data () {
+    return {
+      defaultSettings: {
+        selector: '.js-carousel',
+        duration: 300,
+        easing: 'ease',
+        loop: true,
+        perPage: {
+          992: 3,
```
Comment (pawel-schmidt) [inline components/ChapterBlocks.vue:86]: What is this prop for?

File: components/ChapterBlocks.vue [hunk 7]
```diff
@@ -54,13 +69,54 @@ export default {
...
-        .filter(chapter => chapter.level === 2)
-        .slice(0, this.limitSectionInChapter)
+  mounted () {
+    if (this.hasCarousel) {
+      this.initCarousel()
+    }
+  },
+
+  beforeDestroy () {
+    if (this.hasCarousel) {
```
Comment (pawel-schmidt) [inline components/ChapterBlocks.vue:99]: How about this flow:
1. Component is mounted with `:hasCarousel="hasCarousel"`, where `hasCarousel = true`;
2. `hasCarousel` changes to `false`;
3. Component is going to be destroyed.

Isn't a memory leak here?

Comment (grzegorz-s) [reply components/ChapterBlocks.vue:99]: you use those components in `.md` files only where the activity is not the case, once set, it  stays

File: components/ChapterBlocks.vue (hunk 6 above)
Comment (grzegorz-s) [reply components/ChapterBlocks.vue:86]: that config for carousel, from 992px screen width, display 3 items, default is 1

File: components/ChapterBlocks.vue (hunk 4 above)
Comment (grzegorz-s) [reply components/ChapterBlocks.vue:51]: haha, not the first time I use it 😅 supporting polish made

File: components/ChapterBlock.vue (hunk 1 above)
Comment (grzegorz-s) [reply components/ChapterBlock.vue]: moved it to data

File: components/ChapterBlocks.vue (hunk 4 above)
Comment (wooojek) [reply components/ChapterBlocks.vue:51]: 🇵🇱

File: components/ChapterBlocks.vue (hunk 6 above)
Comment (grzegorz-s) [reply components/ChapterBlocks.vue:88]: and I've changed naming a bit for `defaultCarouselSettings:`  - should be clearer now

**Pull request merged**